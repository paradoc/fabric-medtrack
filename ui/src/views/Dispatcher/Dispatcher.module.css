.container {
  width: 100vw;
  display: grid;
  grid-template-columns: auto 1fr;
}

.container > aside {
  width: 250px;
  color: white;
  background-color: #444;
  display: grid;
  grid-template-rows: 1fr min-content;
  justify-items: center;
  padding: 24px 0;
}

.container > aside > section {
  width: 100%;
  display: grid;
}

.meta {
  grid-template-rows: min-content min-content;
  row-gap: 48px;
}

.meta > div {
  display: grid;
}

.user {
  row-gap: 6px;
  width: 100%;
}

.meta > div:last-of-type,
.user > * {
  padding: 0 24px;
}

.user > strong {
  font-weight: bold;
  font-size: 24px;
}

.user > span {
  font-size: 16px;
  font-weight: 100;
}

.buttons {
  display: grid;
  place-content: center;
}

.buttons > button {
  width: 148px;
  height: 48px;
  color: white;
  background: none;
  border: none;
  font-weight: bold;
  font-size: 16px;
  text-transform: uppercase;
}

.main {
  padding: 24px 64px;
  row-gap: 48px;
  display: grid;
  grid-template-rows: auto 1fr;
}