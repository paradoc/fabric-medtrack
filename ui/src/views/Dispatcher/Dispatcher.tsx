import React, { useEffect, useState } from 'react'
import { Link, Navigate, Outlet } from 'react-router-dom'
import dayjs from 'dayjs'
import { useSessionStorage } from 'usehooks-ts'

import Logo from '../../components/Logo'

import styles from './Dispatcher.module.css'

function CurrentTime() {
  const [time, setTime] = useState(dayjs().format('YYYY-MM-DD HH:mm:ss'))

  useEffect(() => {
    const id = setInterval(() => {
      setTime(dayjs().format('MMM D, YYYY HH:mm:ss ddd'))
    }, 1000)
    return () => {
      clearInterval(id)
    }
  }, [])

  return <div className={styles.time}>{time}</div>
}

export default function Dispatcher() {
  const [isLoggedIn, setIsLoggedIn] = useSessionStorage('login', false)

  return !isLoggedIn ? (
    <Navigate to="/login" />
  ) : (
    <div className={styles.container}>
      <aside>
        <section className={styles.meta}>
          <div className={styles.user}>
            <strong>Mark C.</strong>
            <span>RPh</span>
          </div>
          <CurrentTime />
        </section>
        <section className={styles.buttons}>
          <button onClick={() => setIsLoggedIn(false)} data-testid="logout">
            Logout
          </button>
        </section>
      </aside>
      <main className={styles.main}>
        <Link
          to="/dispatcher"
          style={{ textDecoration: 'none', color: '#333' }}
        >
          <Logo />
        </Link>
        <Outlet />
      </main>
    </div>
  )
}
