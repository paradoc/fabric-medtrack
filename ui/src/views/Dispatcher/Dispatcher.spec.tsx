import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import Dispatcher from './Dispatcher'
import { MemoryRouter } from 'react-router'

jest.mock('usehooks-ts', () => ({
  ...jest.requireActual('usehooks-ts'),
  useSessionStorage: () => [true, () => {}],
}))

describe('Dispatcher', () => {
  beforeAll(() => {
    jest.useFakeTimers().setSystemTime(new Date('2022-12-01'))
  })
  afterAll(() => {
    jest.clearAllTimers()
  })
  test('should match snapshot', () => {
    const { container, getByTestId } = render(
      <MemoryRouter>
        <Dispatcher />
      </MemoryRouter>
    )
    expect(container).toMatchSnapshot()
    const logoutButton = getByTestId('logout')
    fireEvent.click(logoutButton)
  })
})
