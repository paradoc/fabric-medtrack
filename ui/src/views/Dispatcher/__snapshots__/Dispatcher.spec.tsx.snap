// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dispatcher should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <aside>
      <section
        class="meta"
      >
        <div
          class="user"
        >
          <strong>
            Mark C.
          </strong>
          <span>
            RPh
          </span>
        </div>
        <div
          class="time"
        >
          2022-12-01 09:00:00
        </div>
      </section>
      <section
        class="buttons"
      >
        <button
          data-testid="logout"
        >
          Logout
        </button>
      </section>
    </aside>
    <main
      class="main"
    >
      <a
        href="/dispatcher"
        style="text-decoration: none; color: rgb(51, 51, 51);"
      >
        <div
          class="logo"
        >
          <strong>
            Mars
          </strong>
          <span>
            Drugstore
          </span>
        </div>
      </a>
    </main>
  </div>
</div>
`;
